src/viam/sdk/robot/service.cpp

```diff
@@ -160,14 +160,23 @@ ::grpc::Status RobotService_::StopAll(::grpc::ServerContext* context,
         const ResourceName rn = resource->get_resource_name(resource->name());
         const std::string rn_ = rn.SerializeAsString();
         if (extra.find(rn_) != extra.end()) {
-            grpc::StatusCode status = resource->stop(extra.at(rn_));
-            if (status != grpc::StatusCode::OK) {
-                // NOLINTNEXTLINE(clang-analyzer-deadcode.DeadStores)
-                status = resource->stop();
+            try {
+                resource->stop(extra.at(rn_));
+            } catch (...) {
+                try {
+                    // NOLINTNEXTLINE(clang-analyzer-deadcode.DeadStores)
+                    resource->stop();
+                } catch (...) {
+                    status = grpc::UNKNOWN;
+                }
             }
         } else {
-            // NOLINTNEXTLINE(clang-analyzer-deadcode.DeadStores)
-            status = resource->stop();
+            try {
+                // NOLINTNEXTLINE(clang-analyzer-deadcode.DeadStores)
+                resource->stop();
+            } catch (...) {
+                status = grpc::UNKNOWN;
+            }
```

Comment:
logic here should be pretty much the same, but have to do nested `try/catch` blocks in order to attempt `stop` both with and without an `extra`. Seems slightly clunky but simple enough that I was okay with it, happy to revisit if folks don't love this implementation!

Reply:
I guess my only issue with this is that we'll never get the error from the `stop(extra)` call, since this code always assumes the issue was that we can't use `extra`, and not that there was some other gRPC/resource error.

Reply:
Fair concern; I suppose that issue already existed. Looks like Python [only catches type errors](https://github.com/viamrobotics/viam-python-sdk/blob/main/src/viam/robot/service.py#L149); I suppose we could make the catch on L165 more granular than `...`.

Reply:
Yeah definitely a valid concern! This was already a bit of a concern (we were before only capturing the status code of the error, not the error message itself). At this point, making sure we return the status code becomes pretty complicated but we _can_ update to return the text of the error when something fails. I made a quick change to that effect; it'll only keep track of the latest error message seen but seems better than nothing? Let me know what you think!

As far more granular matching on the kind of error, I'd propose holding off on that until we implement our custom `ViamError` type.